### DISCUSSION TITLE
OPENDTU und Wechselrichter Hoymiles HMS 1600 4T

### DISCUSSION THREAD

**Discussion by WotanValhalla** [tbnobody/OpenDTU#1599] [General]
Mahlzeit.

Ich habe mal eine Frage an die wissenden.
Ich habe ein OPENDTU Rev 2.1 von KiCad.
Auf dem Board befindet sich eine NRF24L01 antenne.
Ich habe einen Händler gefunden, der die Antenne CMT2300A (natürlich mit dem Board dazu) verkauft.
Wenn ich jetzt die NRF Antenne runter löte und dann die CMT Antenne dran löte, ist es dann möglich meinen Wechselrichter der Serie HMS 4T 1600 damit auszulesen?
Denn derzeit wird der zwar erkannt von der NRF Antenne, aber die Auslesedaten sind immer auf 0.
Oder habe ich da gar was falsch gemacht?
Für mich ist das alles Neuland, aber wer nicht fragt, wird doof sterben, hehe.

**Comment by Juergen2453** [comment]
Dort kannst du nachlesen welches RF Modul für welches Model du brauchst. für den HMS-1600 4T brauchst du das Modul CMT2300A mit der richtigen Antenne.

[https://github.com/tbnobody/OpenDTU](https://github.com/tbnobody/OpenDTU)

**Reply by WotanValhalla** [reply]
Wenn ich das richtig verstanden habe, heisst die Antenne CMT2300A die ich brauche.
Das Board auf dem der opendtu läuft und die Antenne sind 2 unterschiedliche paar Schuhe.
Und meine Frage war ja daher, ob ich einfach die Originale Antenne die ich bei meinem Kompletten OPENDTU mit der Antenne NRF behalten kann und lediglich die Antenne NRF gegen die Antenne CMT austsuschen kann.

**Comment by Juergen2453** [comment]
Du hast ein Board auf dem OpenDTU läuft.
Dann hast du noch ein Funk-Modul "CMT2300A oder NRF24L01+"
Für den HMS1600 brauchst du das Funk-Modul CMT2300A.
Der NRF24L01+ sendet auf der Frequenz "2,4Ghz"
Der CMT2300A sendet auf  der Frequenz "868Mhz"
Die Antennen dazu (sind ja an die Frequenzen angepasst) sollten natürlich zu den Funk-Modulen passen.
Zu jedem Funk-Modul werden in der Regel auch die passenden Antennen mitgeliefert.

P.S. Du schreibst immer "...heisst die Antenne CMT2300A die ich ..." . Das Funk-Modul heißt so.

Du kannst sogar Glück haben das eine einfache Büroklammer als Antenne läuft.

<img width="389" alt="image" src="https://github.com/tbnobody/OpenDTU/assets/116970574/d0e6352a-ef94-43fc-b9fb-63b044472095">

**Reply by WotanValhalla** [reply]
Hallo Jürgen.
Jetzt verstehe ich was du meinst, hehe.
Naja, ich meinte natürlich das Bord inkl. Antenne, smile...

Aber um das ich das richtig verstehe, ich könnte mir jetzt das BORD inkl. Antenne CMT2300A kaufen und das anstelle des NRF24 dran löten und dann hätte ich das, was ich eigentlich brauche für meinen HMS 1600T. 

Richtig?

Gruß...

**Comment by Juergen2453** [comment]
Im Prinzip Ja, nur die PIN Belegung ist eine andere.
Wenn du noch nichts gekauft hast dann schaue mal hier.
Das Board ist hier in der Community Entwickelt worden, und hat alles an Board NRF24+ und CMT2300A.

https://shop.allianceapps.io/products/opendtu-fusion-community-edition

Die Documentation dazu

https://github.com/markusdd/OpenDTUFusionDocs

**Comment by stefan123t** [comment]
Ich habe auch noch so ein Board hier rumliegen: OpenDTU rev 1.0.0-2022, Rückseite Designed with KiCAD Logo.
Da sind alle Pin MISO, MOSI, SCK, CSN, IRQ und CE (plus GND, Vcc) sowohl am 8 pin NRF Sockel als auch separat rausgeführt.

Um statt dem NRF24L01+ Modul das CMT2300A Breakout Board z.B. von BlinkyParts zu montieren,
musst Du entweder das Modul separat verdrahten oder das NRF Modul abstecken / auslöten und stattdessen das CMT2300A Breakout Board einlöten. Die beiden Module sind m.W. Pin-kompatibel, d.h. ja, man kann sie austauschen.

OpenDTU braucht dann noch die Information dass jetzt ein CMT2300A anstelle des NRF24L01+ dran hängt.
Das geht wie in der Doku beschrieben über die pin_mapping.json unter Settings > Device Manager.

Hast Du denn nur den HMS-1600-4T und keinen HM-Wechselrichter, der den NRF24L01+ benötigt ?
Für einen Parallelbetrieb beider Funkmodule musst Du m.W. IRQ, CE und CSN evtl. separat verdrahten, damit OpenDTU die Module dediziert ansprechen kann. Genaues findet sich aber zur Not im Discord oder in den Schaltplänen hier auf github.
